lib/wakes.rb

```diff
@@ -39,8 +39,12 @@ def self.create!(*args)
     build(*args).tap(&:save!)
   end
 
-  def self.redirect(source, target, label = nil)
-    RedirectMapper.new(source, target, label)
+  def self.redirect(source_path, target_path, label = nil)
+    RedirectMapper.new(source_path, target_path, label)
+  end
+
+  def self.redirect_with_hosts(source_host: nil, source_path:, target_host: nil, target_path:, label: nil)
```

Comment:
@chrismaddocks what do you mean by "quasi-URI syntax"?

I wouldn't start from the syntax that you've been using elsewhere. I would start from the question of, "what's the ideal API for this?" and then figure out how to implement it. You probably should support a few different cases:

    Wakes.redirect '/source', '/target'
    Wakes.redirect 'www.domain.com/source', '/target'
    Wakes.redirect 'http://www.domain.com/source', '/target'